### PR TITLE
Use safe-buffer & License, drop Node<4 and md5.js

### PR DESCRIPTION
**Pull request by dcousens**

### CODE REVIEW

File: make-hash.js [hunk 1]
```diff
@@ -1,30 +1,32 @@
...
-var zeroBuffer = new Buffer(intSize)
-zeroBuffer.fill(0)
+var INT32_SIZE = 4
+var CHAR64_SIZE = 8
+var HASH128_SIZE = 16
 
-var charSize = 8
-var hashSize = 16
+var Buffer = require('safe-buffer').Buffer
+var zeroBuffer = Buffer.alloc(INT32_SIZE, 0)
```
Comment (calvinmetcalf) [inline make-hash.js]: alloc zero fills by default

Comment (dcousens) [inline make-hash.js]: I know,  it was just about making that clear

**Comment by dcousens** [comment]
added LICENSE, and some other cleanup

**Comment by dcousens** [comment]
Added `md5.js` and latest RIPEMD160,  dropped `safe-buffer` requirement entirely.  Will update those repos now.

Good to merge.

**Comment by dcousens** [comment]
@fanatid it seems I've repeated your work in #13

**Pull request merged**